### PR TITLE
New custom calibration STIS notebook

### PR DESCRIPTION
**Pull request by jotaylor**
I also had to update a few small things in the python scripts, to handle the NB runs.

### CODE REVIEW

File: notebooks/ullyses-custom-stis.ipynb [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To rectify the data, we must reprocess it using the custom ULLYSES wrapper around CalSTIS. This wrapper can perform any number of corrections as specified in an input YAML file. This includes: \n",
+    "- Removing the fringe pattern. Using the [`stistools` defringing](https://stistools.readthedocs.io/en/latest/defringe_guide.html) package, the amplitude, shift, and position of the fringes are determined for each science spectrum. This is done by processing the contemporaneous fringe flat taken right before or after each G750L or G750M science exposure.\n",
+    "- Manually calculating the number of pixels with high dark rate (DQ=16), using a modified threshold.\n",
+    "- Performing a custom extraction to one or more targets in the long slit. This is done e.g. when the target is especially faint and requires a smaller extraction box, when the target is miscentered in the slit, or there is a companion source in the slit.\n",
+    "\n",
+    "Let's reprocess the SZ-76 data by using the YAML files as input to the ULLYSES code."
```
Comment (Rplesha) [inline notebooks/ullyses-custom-stis.ipynb:441]: Make a quick note here that there are plots produced by the function after the code finishes running, but we'll look at those in more detail in the sections below.

File: notebooks/ullyses-custom-stis.ipynb [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+    "from IPython.display import display_pdf\n",
+    "with open('custom_products/sz-76_g750l_oeimds030_1d_compare.pdf', \"rb\") as f:\n",
+    "    display_pdf(f.read(),raw=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A few things to note: the improvement for G230L is minimal to nonexistent, but there is a marked difference for the CCD datasets. For both G430L and G750L, then number of spectral elements marked as having a serious DQ flag has dramatically decreased. This means that the ULLYSES coaddition algorithm will no longer have the large number of dropouts we saw before. For G750L, the NIR fringing has been successfully removed.\n",
```
Comment (Rplesha) [inline notebooks/ullyses-custom-stis.ipynb:543]: remind people that they are looking at the Xs for this purpose

**Review by Rplesha — COMMENTED** [review]
I think things look really good! With the exception of the numpy import, I am able to run the notebook fully.

There are a few places where I think a little bit more detail could be helpful that I've noted here, but I think it's really great as it's written now!

File: notebooks/ullyses-custom-stis.ipynb [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+    "import sys\n",
+    "current_stdout = sys.stdout\n",
+    "import os\n",
+    "import glob\n",
+    "import pprint\n",
+    "from collections import Counter\n",
+    "\n",
+    "from astropy.io import fits\n",
+    "\n",
+    "%matplotlib inline\n",
```
[the remark this answers is not in the document]
Comment (jotaylor) [reply notebooks/ullyses-custom-stis.ipynb:109]: I couldn't ever get this to work on my system... 😢 I don't know what the issue is. I get crazy javascript error every time I run it with inline.

[the remark this answers is not in the document]
Comment (Rplesha) [reply notebooks/ullyses-custom-stis.ipynb:109]: We have just the widget and not inline. Could that be the issue?

**Comment by jotaylor** [comment]
@Rplesha thanks for the review!! There are two things I'd like to follow up on. Is the fringing explanation more clear now? I kept the order as is, cause I want to display the plot before talking about what we see in it, but I added more details about what causes it. 

As for matplotlib, I replaced my inline call with widget and whenever I make plots, it breaks my notebook :/ Not sure what to do about it.

**Review by Rplesha — APPROVED** [review]
Yes I think it looks great! The fringing makes a lot more sense now and I can see at the bottom how much it's improved, too.

I'm not too worried about the widget stuff. I think it is useful to be able to zoom into plots, but it isn't a showstopper. Perhaps just having it as a commented out version as an option instead?

**Pull request merged**